src/js/midi/midi.js

```diff
@@ -165,74 +163,76 @@ class MIDI extends EventEmitter {
             }
           }
         }
-      })
+      }
 
       if (navigator.requestMIDIAccess) {
         const midiAccess = await navigator.requestMIDIAccess({ sysex: false })
         enableMidiSupport(midiAccess)
-        this.emit('ready', R.clone(this._.status))
+        this.emit('ready')
       } else {
-        this.emit('blocked', R.clone(this._.status))
+        this.emit('blocked')
       }
     }
   }
 
-  toggleDevice(type, name) {
-    const { status } = this._
+  toggleDevice(type, name, newValue = null) {
+    const { devices } = this._
 
-    const device = status.devices[`${type}s`][name]
-    device.enabled = !device.enabled
+    const device = devices[`${type}s`][name]
+    device.enabled = newValue === null ? !device.enabled : newValue
 
     if (type === 'output') {
       if (device.enabled) {
-        R.forEach((channel) => {
+        device.channels.forEach((channel) => {
           device.port.send(setPitchBendLimit(channel, maxBendingDistanceInSemitones))
-        })(device.channels)
+        })
       } else {
-        R.forEach((channel) => {
+        device.channels.forEach((channel) => {
           device.port.send(bendPitch(channel, 0))
-        })(device.channels)
+        })
       }
     }
 
-    this.emit('update', R.clone(status))
+    this.emit('update')
   }
 
-  toggleChannel(type, name, channelID) {
-    const { status } = this._
+  setDevice(type, name, newValue) {
+    this.toggleDevice(type, name, newValue)
+  }
+
+  toggleChannel(type, name, channelID, newValue = null) {
+    const { devices } = this._
 
-    const device = status.devices[`${type}s`][name]
+    const device = devices[`${type}s`][name]
+    const channel = device.channels.find(({ id }) => id === channelID)
 
-    if (device.enabled) {
-      const channel = R.find(R.propEq('id', parseInt(channelID)))(device.channels)
-      channel.enabled = !channel.enabled
-      this.emit('update', R.clone(status))
+    newValue = newValue === null ? !channel.enabled : newValue
+    if (channel.enabled !== newValue) {
+      channel.enabled = newValue
+      this.emit('update')
     }
   }
 
+  setChannel(type, name, channelID, newValue) {
+    this.toggleChannel(type, name, channelID, newValue)
+  }
+
   getEnabledOutputs() {
-    return R.compose(
-      R.filter((device) => device.enabled && R.any((channel) => channel.enabled)(device.channels)),
-      R.values
-    )(this._.status.devices.outputs)
+    return Object.values(this._.devices.outputs).filter(({ enabled, channels }) => {
+      return enabled === true && channels.find(({ enabled }) => enabled === true) !== undefined
+    })
   }
 
   getLowestEnabledChannel(channels) {
-    return R.find((channel) => channel.enabled, channels)
+    return channels.find(({ enabled }) => enabled === true)
   }
 
-  // -------------------------------------------
-
   playFrequency(frequency = 0, noteLength = Infinity) {
```

Comment:
This may be beyond the scope of this PR, but if I'm reading the code correctly a finite `noteLength` turns off all notes and not just the one started by this call. Doesn't feel like the intended behavior.